### ISSUE TITLE
`cmp_owned` applied to PathBuf::from converts to non-compileable code

### ISSUE THREAD

**Issue by daniel-levin** [rust-lang/rust-clippy#16897]
### Summary

See reproducer, it is fairly self-explanatory.

### Reproducer

Code:

```rust
use std::path::PathBuf;

fn main() {
    let p = PathBuf::from("abc");

    if p == PathBuf::from("cde") {
        panic!("pain");
    }
}

```

Current output:
```                                                                                                                                                                                                                                                                                                                                             
after fixes were automatically applied the compiler reported errors within these files:

  * src/main.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust-clippy/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0277]: can't compare `std::path::PathBuf` with `&str`
 --> src/main.rs:6:10
  |
6 |     if p == "cde" {
  |          ^^ no implementation for `std::path::PathBuf == &str`
  |
  = help: the trait `std::cmp::PartialEq<&str>` is not implemented for `std::path::PathBuf`
help: consider dereferencing both sides of the expression
  |
6 |     if *p == *"cde" {
  |        +     +

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0277`.

```

Desired output:
Fixes the lint

### Version

```text
❯ rustc -Vv
rustc 1.94.1 (e408947bf 2026-03-25)
binary: rustc
commit-hash: e408947bfd200af42db322daf0fadfe7e26d3bd1
commit-date: 2026-03-25
host: x86_64-unknown-linux-gnu
release: 1.94.1
LLVM version: 21.1.8


fwiw repros on macOS, illumos
```

### Additional Labels

_No response_

**Comment by profetia** [comment]
Already fixed in #16628
